java.hints/src/org/netbeans/modules/java/hints/jdk/ConvertVarToExplicitType.java

```diff
@@ -0,0 +1,170 @@
+     *
+     */
+    private static final class JavaFixImpl extends JavaFix {
+
+        public JavaFixImpl(CompilationInfo info, TreePath tp) {
+            super(info, tp);
+        }
+
+        @Override
+        @Messages("FIX_convertVarToExplicitType=Replace var with explicit type")
+        protected String getText() {
+            return Bundle.FIX_convertVarToExplicitType();
+        }
+
+        @Override
+        protected void performRewrite(TransformationContext tc) throws Exception {
+            WorkingCopy wc = tc.getWorkingCopy();
+            CompilationUnitTree cut = wc.getCompilationUnit();
+            TreePath statementPath = tc.getPath();
+
+            TreeMaker make = wc.getTreeMaker();
+
+            if (statementPath.getLeaf().getKind() == Tree.Kind.VARIABLE) {
+                VariableTree oldVariableTree = (VariableTree) statementPath.getLeaf();
+                TypeMirror type = wc.getTrees().getTypeMirror(statementPath);
+                VariableTree newVariableTree = make.Variable(
+                        oldVariableTree.getModifiers(),
+                        oldVariableTree.getName(),
+                        make.Type(type),
+                        oldVariableTree.getInitializer()
+                );
+                wc.rewrite(oldVariableTree, newVariableTree);
+            }
+        }
+
+    }
+
+    /**
+     *
+     * @param ctx : HintContext
+     * @return true if pre-conditions for hint to be enable is meet
+     */
+    private static boolean isLocalVarType(HintContext ctx) {
+
+        CompilationInfo info = ctx.getInfo();
+        
+        if (info.getSourceVersion().compareTo(SourceVersion.RELEASE_9) < 1) {
+            return false;
+        }        
+
+        TreePath treePath = ctx.getPath();
+
+        // should be local variable
+        if (info.getTrees().getElement(treePath).getKind() != ElementKind.LOCAL_VARIABLE) {
+            return false;
+        }
+
+        // variable declaration of type 'var'
+        return info.getTreeUtilities().isVarType(treePath);
+    }
+
+    private static boolean hasDiagnosticErrors(CompilationInfo info, Tree tree) {
+        long startPos = info.getTrees().getSourcePositions().getStartPosition(info.getCompilationUnit(), tree);
+        long endPos = info.getTrees().getSourcePositions().getEndPosition(info.getCompilationUnit(), tree);
+
+        for (Diagnostic<?> d : info.getDiagnostics()) {
+            if (d.getKind() == Kind.ERROR) {
+                if ((d.getPosition() >= startPos) && (d.getPosition() <= endPos)) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    //filter anonymous class and intersection types
+    private static boolean isValidType(HintContext ctx) {
+        TreePath treePath = ctx.getPath();
+        TreePath initTreePath = ctx.getVariables().get("$init");  //NOI18N
+
+        if (initTreePath.getLeaf().getKind() == Tree.Kind.NEW_CLASS) {
```

Comment:
So this will filter out:
var v = new Object() {};

That's OK/as intended at this time. But the following won't be filtered out, unless I am missing something:
<Z> Z get(Z z) { return z; }

var v = get(new Object() {});

I guess I am fine if fixing this is deferred, but seems we will need to inspect the type, including type parameters, for anonymous types.

Reply:
Logged https://issues.apache.org/jira/browse/NETBEANS-774 to track this issue.